code/test_planning_top/behavior_agent/readme.md

```diff
@@ -0,0 +1,157 @@
+The following description is not complete, it just contains the most common behaviours and subtrees. For a complete description have a look at the [tree-description](https://github.com/ll7/psaf2/blob/main/documentation/behaviour_agent/behaviortree.xml)
+(you can open it with [Groot](https://github.com/BehaviorTree/Groot)) and the [bt-specs](https://github.com/ll7/psaf2/blob/main/documentation/BTSpecs.md).
+Note that we didnt actually implement all of the behaviours from this design, due to time limitations.
+
+### Legend
+
+The following Notation is used in this documentation:
+![BT Legend](https://github.com/ll7/psaf2/blob/main/documentation/behaviour_agent/bt-legend.svg)
+
+### Big Picture
+
+![BT Big Picture](https://github.com/ll7/psaf2/blob/main/documentation/behaviour_agent/bt_big_picture.svg)
+This top-level tree consists mainly of sub-trees that are explained below. If none of the subtrees fit the current situation, the behaviour_agent goes into _Cruising_-behaviour, where it just follows the Path at an appropriate speed.
+
+### Intersection
+
+![BT Intersection](https://github.com/ll7/psaf2/blob/main/documentation/behaviour_agent/bt-intersection.svg)
+If there is a Intersection coming up the agent executes the following sequence of behaviours:
+
+* Approach Intersection
+
+    Slows down, gets into the right lane for turning and stops at line
+* Wait at Intersection
+
+    Waits for traffic lights or higher priority traffic
+* Enter Intersection
+
+    Enters the intersection and stops again, if there is higher priority oncoming traffic
+* Leave Intersection
+
+    Leaves the intersection in the right direction
+
+### Roundabout
+
+![BT Roundabout](https://github.com/ll7/psaf2/blob/main/documentation/behaviour_agent/bt-roundabout.svg)
+This subtree is basically identical to the intersection-subtree. The implementation of the behaviours varies a bit though.
+
+### Overtaking
+
+The Overtaking subtree is quite big to accommodate for different overtaking scenarios.
+Please have a look at the [tree-description](https://github.com/ll7/psaf2/blob/main/documentation/behaviour_agent/behaviortree.xml) and the [bt-specs](https://github.com/ll7/psaf2/blob/main/documentation/BTSpecs.md) for a more detailed description. The Multi-Lane Overtaking Subtree looks like this:
+![BT Overtaking](https://github.com/ll7/psaf2/blob/main/documentation/behaviour_agent/bt-overtaking.svg)
+
+* Multi Lane?
+
+    Checks the map data: does the current road have more than one lane?
+* Left Lane available?
+
+    Checks the map data: is there a lane available to the left of the ego vehicle?
+* Wait for Left Lane free
+
+    Waits for the left lane to be free. This has a timeout.
+* Switch Lane Left
+
+    Triggers a Laneswitch to the left by calling the local planner
+
+### Right-Hand Driving
+
```

Comment:
It might be useful to have a option to abort overtaking if it we can't get past the car in front fast enough

Reply:
This will be done, the old readme is only included to have a starting point for the documentation later on